### PR TITLE
fix: [CDS-76068]: Environment Edit on pipeline studio still shows vars, manifests, configFiles with Override V2 enabled

### PR DESCRIPTION
**Pull request by MondalSayantan**
### Summary

`EnvironmentDetails.tsx` which is the parent of `EnvironmentConfiguration.tsx` and also responsible for passing the boolean value for the feature flag is not being rendered in environment edit on pipeline studio so now the prop is passed from `DeployEnvironments.tsx`

#### Screenshots

https://github.com/harness/harness-core-ui/assets/62999423/8425c546-6bdd-44fa-8c33-f29d960ab66a

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [x] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by MondalSayantan** [comment]
retrigger cypress-pipeline

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/EnvironmentConfiguration/__tests__/EnvironmentConfiguration.test.tsx [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+                isEdit={true}
+                isModified={false}
+                setYamlHandler={jest.fn()}
+              />
+            </FormikForm>
+          )}
+        </Formik>
+      </TestWrapper>
+    )
+    await waitFor(() => expect(container).toMatchSnapshot())
```
Comment (samarth-sharma31) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/EnvironmentConfiguration/__tests__/EnvironmentConfiguration.test.tsx:130]: Let's try to avoid snapshot testing whenever possible. Please replace it with assertions like ->

expect(**someComponent**).toBeDefined()

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/EnvironmentConfiguration/__tests__/EnvironmentConfiguration.test.tsx [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+                isEdit={true}
+                isModified={false}
+                setYamlHandler={jest.fn()}
+              />
+            </FormikForm>
+          )}
+        </Formik>
+      </TestWrapper>
+    )
+    expect(container).toMatchSnapshot()
```
Comment (samarth-sharma31) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/EnvironmentConfiguration/__tests__/EnvironmentConfiguration.test.tsx:228]: https://github.com/harness/harness-core-ui/pull/17261/files#r1285236160

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironment/DeployEnvironment.tsx [hunk 3]
```diff
@@ -127,12 +128,32 @@ export default function DeployEnvironment({
...
+  const isServiceOverridesEnabled = CDS_SERVICE_OVERRIDES_2_0 && enableServiceOverrideSettings?.data?.value === 'true'
+  React.useEffect(() => {
+    if (enableServiceOverrideSettingsError) {
+      showError(getRBACErrorMessage(enableServiceOverrideSettingsError))
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [enableServiceOverrideSettingsError])
 
-  const { CD_NG_DYNAMIC_PROVISIONING_ENV_V2, CDS_SERVICE_OVERRIDES_2_0: isOverridesEnabled } = useFeatureFlags()
+  // mock show error
```
Comment (samarth-sharma31) [inline src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironment/DeployEnvironment.tsx:155]: Please remove unwanted comments

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironment/DeployEnvironment.tsx [hunk 4]
```diff
@@ -127,12 +128,32 @@ export default function DeployEnvironment({
...
+  //.toHaveBeenCalled
```
Comment (samarth-sharma31) [inline src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironment/DeployEnvironment.tsx:156]: https://github.com/harness/harness-core-ui/pull/17261/files#r1285236338

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/EnvironmentEntitiesList/EnvironmentEntitiesList.tsx [hunk 5]
```diff
@@ -122,7 +124,7 @@ export default function EnvironmentEntitiesList({
         enforceFocus={false}
         lazy
         width={1128}
-        height={840}
+        height={isServiceOverridesEnabled ? 600 : 840}
```
Comment (samarth-sharma31) [inline src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/EnvironmentEntitiesList/EnvironmentEntitiesList.tsx:127]: Why is this required ?

Comment (MondalSayantan) [reply src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/EnvironmentEntitiesList/EnvironmentEntitiesList.tsx:127]: this adjusts the modal height depending on `isServiceOverrridesEnabled`

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/AddEditEnvironmentModal.tsx [hunk 6]
```diff
@@ -51,6 +51,7 @@ export interface AddEditEnvironmentModalProps {
   isEdit: boolean
   onCreateOrUpdateInsideGroup?: (newEnv: EnvironmentResponse) => void
```
Comment (samarth-sharma31) [inline src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/AddEditEnvironmentModal.tsx:52]: Please attach screenshots and video of the behaviour - this is mandatory for all PRs which have any possible UX changes

**Review by samarth-sharma31 — APPROVED** [review]

**Review by Sagar-Rajput7 — APPROVED** [review]
